### ISSUE TITLE
Allow controlling whether the soname is set in cc_shared_library.

### ISSUE THREAD

**Issue by oquenchil** [bazelbuild/bazel#18798]
Add new toolchain feature to allow the soname to be added to cc_shared_libraries when the feature is enabled.

The following condition (https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/cpp/CcLinkingHelper.java;drc=044a14cca2747aeff258fc71eaeb153c08cb34d5;l=687)  should have: 
```
|| (dynamicLinkType == LinkTargetType.DYNAMIC_LIBRARY && featureConfiguration.isEnabled("add_soname_to_transitive_shared_libraries")
```

The feature "add_soname_to_transitive_shared_libraries" doesn't exist yet and needs to be added to the default unix_cc_toolchain_config.bzl toolchain.

This is an incompatible change and projects would have to explicitly enable the feature for all their shared libraries or individual ones. This is an incompatible change because existing builds copy and rename the cc_shared_library library output, if the soname is set then whatever links against it will use the soname and the library won't be found at runtime.

A workaround is to pass the flag `-Wl,-soname=libA.so` via `cc_shared_library.user_link_flags`.

_Originally posted by @Distancess in https://github.com/bazelbuild/bazel/discussions/18765#discussioncomment-6299523_

**Comment by keith** [comment]
https://github.com/bazelbuild/bazel/pull/23839

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!
